### PR TITLE
Added options to update the reserved memory and the reserved cpu(in MH…

### PR DESCRIPTION
**Pull request by filemoncarrillo**
##### ISSUE TYPE
- Feature Pull Request
##### COMPONENT NAME

-ansible-core/cloud/vmware/vsphere_guest.py
##### ANSIBLE VERSION

```
ansible 1.9.4
```
##### SUMMARY

Need to add an option to update the reserved memory and reserved cpu on vmware vms. This can be done to an existing vm using the reconfigure option of this module. The logic is very similar to the way memory and cpu are assigned so they are in the same block.

```
TASK: [update reserved cpu and reserved memory on  vms] **************************************** 
changed: [cobra18]

PLAY RECAP ******************************************************************** 
cobra18                    : ok=3    changed=1    unreachable=0    failed=0   

```

Sample call to this module that will update reserved fields. 
    - name: update reserved cpu and reseved memory on vms 
      vsphere_guest:
        vcenter_hostname: "{{ vcenter }}"
        username: "{{ vcenter_user }}"
        password: "{{ vcenter_pass }}"
        guest: "{{ inventory_hostname }}" 
        state: reconfigured
        force: true 
        vm_extra_config:
          vcpu.hotadd: true 
          mem.hotadd: true 
          notes: "Updating memory and cpu"
        vm_hardware:
          memory_mb: "{{ memory }}"
          num_cpus: "{{ cpus }}"
          scsi: paravirtual
          reserved_memory: "{{ reserved_memory }}"
          reserved_cpu: "{{ reserved_cpu }}"

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @filemoncarrillo for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by filemoncarrillo** [comment]
Thanks @gregdek for looking at this. I believe I am now:
'ready_for_review'

Filemon Carrillo

**Comment by gregdek** [comment]
Thanks @filemoncarrillo. To the current maintainers, @chrrrles, @rhoop please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by chrrrles** [comment]
As we are now setting these properties, it might be helpful to update gather_facts() so that it reports the values for vm.properties.resourceConfig.cpuAllocation.reservation and vm.properties.resourceConfig.memoryAllocation.reservation

**Comment by gregdek** [comment]
@chrrrles a reminder, if you want a revision, please be sure to use the text "needs_revision" :)

**Comment by chrrrles** [comment]
@gregdek - Ahh, needs_revision  :)  - otherwise +1

**Comment by gregdek** [comment]
Thanks again to @filemoncarrillo for this PR, and thanks @chrrrles, @rhoop for reviewing. Marking for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

File: cloud/vmware/vsphere_guest.py [hunk 1]
```diff
@@ -1507,7 +1549,9 @@ def main():
         'memory_mb': int,
         'num_cpus': int,
         'scsi': basestring,
-        'osid': basestring
+        'osid': basestring,
+	'reserved_memory': int,
```
Comment (resmo) [inline cloud/vmware/vsphere_guest.py]: would you mind fix the indentation?

**Comment by gregdek** [comment]
Thanks @filemoncarrillo for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by filemoncarrillo** [comment]
@gregdek I fixed a suggested indent issue and now Travis CI is failing,  I am not sure how to address it. Can you help me get thru this. Thank you. If change is causing trouble, please just revert it to version that was ready to shipit. Thank you.

**Comment by resmo** [comment]
Please check if you used tabs "ERROR: indentation contains tabs. line 876 column 0"

**Comment by filemoncarrillo** [comment]
Thanks @gregdek for looking at this. I believe I am now:
'ready_for_review'
Tabs were removed as suggested by @resmo 
Unfortunately I lost the shipit label due to some bad commits

Filemon Carrillo

**Comment by resmo** [comment]
@filemoncarrillo that is okay. this one is close for integration. 

Before I ship, could you add a little note in the doc on https://github.com/filemoncarrillo/ansible-modules-core/blob/devel/cloud/vmware/vsphere_guest.py#L110 to let users know these new options are available since version 2.2.

Did you see the comment of @chrrrles here https://github.com/ansible/ansible-modules-core/pull/3543#issuecomment-214917986 about facts? Has this been addressed?

**Comment by gregdek** [comment]
Thanks @filemoncarrillo. To the current maintainers, @chrrrles, @rhoop please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by chrrrles** [comment]
Seconding @resmo on updating the docstring and gather_facts() to include these new parameters.

**Comment by filemoncarrillo** [comment]
@resmo, @gregdek, @chrrrles  I updated the documentation, I plan on updating the gather facts later.

**Comment by filemoncarrillo** [comment]
@resmo, @gregdek, @chrrrles  We will now display reserved memory and reserved cpu when getting vmguest facts. Yeah!!

File: cloud/vmware/vsphere_guest.py [hunk 2]
```diff
@@ -112,9 +109,12 @@
     default: null
   vm_hardware:
     description:
-      - A key, value list of VM config settings. Must include ['memory_mb', 'num_cpus', 'osid', 'scsi'].
+      - A key, value list of VM config settings. Must include ['memory_mb', 'num_cpus', 'osid', 'scsi', 'reserved_memory', 'reserved_cpu'].
```
Comment (chrrrles) [inline cloud/vmware/vsphere_guest.py]: Ahh - you should note that reserved_memory and reserved_cpu are optional values

File: cloud/vmware/vsphere_guest.py [hunk 3]
```diff
@@ -1409,6 +1457,8 @@ def gather_facts(vm):
         'hw_product_uuid': vm.properties.config.uuid,
         'hw_processor_count': vm.properties.config.hardware.numCPU,
         'hw_memtotal_mb': vm.properties.config.hardware.memoryMB,
+        'hw_reserved_cpu': vm.properties.resourceConfig.cpuAllocation.reservation,
+        'hw_reserved_memory': vm.properties.resourceConfig.memoryAllocation.reservation,
```
Comment (chrrrles) [inline cloud/vmware/vsphere_guest.py:1461]: +1

**Comment by chrrrles** [comment]
Looks good -- although the documentation should clarify that reserved_cpu/memory are optional values.  Aside from that, it helps to squash your commits.

**Comment by filemoncarrillo** [comment]
@gregdek, it has been a while, do you know if my change will make it to 2.2? Do you need to add the shipit label, like you did on May 6th. Thanks

**Comment by gregdek** [comment]
@filemoncarrillo sorry, but it's unlikely that this will make it into 2.2 because we're in feature freeze. After 2.2 releases, I'd recommend attending the Core Team meeting on IRC: https://github.com/ansible/community/blob/master/MEETINGS.md

**Comment by gregdek** [comment]
Thanks @filemoncarrillo for this PR. This module is maintained by the Ansible core team, so it can take a while for patches to be reviewed. Thanks for your patience.

Core team: please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with 'needs_revision' or merge as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
Thanks @filemoncarrillo for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

For help on how to do this cleanly please see http://docs.ansible.com/ansible/community.html#contributing-code-features-or-bugfixes

[This message brought to you by your friendly Ansibull-bot.]

**Pull request closed without merging**